#ATACseq Anaylsis by Jack
## run fastqc to access data quality
```{bash}
set -e
echo ${HOSTNAME} ${SLURM_ARRAY_TASK_ID}

module purge
module load FastQC/0.11.9-Java-11.0.2

fileArray=($(ls /home/gdkendalllab/lab/raw_data/fastq/2023_04_03/JPK*/*))
R1=${fileArray[$SLURM_ARRAY_TASK_ID]}
baseName=${R1##*/}
baseName=${baseName%%_*}

fastqc \
  -o  /home/gdkendalllab/lab/analyses/rsjxk002/P3F_ATAC_6hpf/fastqc \
  ${R1}
```

##remove adapter
```{bash}
set -e
echo ${HOSTNAME} ${SLURM_ARRAY_TASK_ID}

module purge
module load GCCcore/9.3.0
module load cutadapt/2.10-Python-3.8.2

fileArray=($(ls /home/gdkendalllab/lab/raw_data/fastq/2023_04_03/JPK*/*_R1_001.fastq.gz))
R1=${fileArray[$SLURM_ARRAY_TASK_ID]}
R2=${R1/R1_001.fastq.gz/R2_001.fastq.gz}
baseName=${R1##*/}
baseName=${baseName%%_*}

cutadapt \
  -a CTGTCTCTTATA \
  -A CTGTCTCTTATA \
  -m 45 \
  --pair-filter any \
  -o /home/gdkendalllab/lab/analyses/rsjxk002/P3F_ATAC_6hpf/cutadapt/${baseName}_R1_001_cutadapt.fastq.gz \
  -p /home/gdkendalllab/lab/analyses/rsjxk002/P3F_ATAC_6hpf/cutadapt/${baseName}_R2_001_cutadapt.fastq.gz \
  ${R1} \
  ${R2} \
```

##align to zebrafish and index
``` {bash}
set -e
echo ${HOSTNAME} ${SLURM_ARRAY_TASK_ID}

module purge
module load GCC/9.3.0 \
            GCCcore/9.3.0 \
            HISAT2/2.2.1 \
            SAMtools/1.15
            
fileArray=($(ls /home/gdkendalllab/lab/analyses/rsjxk002/P3F_ATAC_6hpf/cutadapt/*_R1_001_cutadapt.fastq.gz))
R1=${fileArray[$SLURM_ARRAY_TASK_ID]}
R2=${R1/R1_001_cutadapt.fastq.gz/R2_001_cutadapt.fastq.gz}
baseName=${R1##*/}
baseName=${baseName%%_*}

hisat2 \
    -x /home/gdkendalllab/lab/references/hisat2/danRer11 \
    -1 ${R1} \
    -2 ${R2} \
    --threads 15 \
    -k 1 \
    --no-spliced-alignment \
    --no-temp-splicesite \
    --no-mixed \
    --no-discordant \
    --summary-file ${baseName}Summary.txt \
    | samtools fixmate -@ 10 -m - - \
    | samtools sort \
        -T /gpfs0/scratch/rsjxk002/ \
        -@ 10 \
        -m 15G \
        -O BAM \
        | samtools view -@ 10 -b - \
    > ${baseName}_align.bam
samtools index ${baseName}_align.bam

samtools markdup -@ 10 -r ${baseName}_align.bam - \
    | samtools view -@ 10 -b - \
    > ${baseName}_dedupd.bam
samtools index ${baseName}_dedupd.bam
```
