---
title: "P3F_ChIP_6hpf_byJack"
author: "Jack Kucinski and Matt Cannon"
date: "`r format(Sys.time(), '%m/%d/%Y')`"
output: 
    html_document:
        toc: true
        toc_float: true
        toc_depth: 5
        number_sections: false
        code_folding: hide
---

# Title P3F_ChIP_6hpf_byJack

## Goals: 
To analyze my first publishable ChIP-seq datasets in parallel with Matt Cannon to gain bioinformatics skills

## Samples
JPK0028	GSL-JK-3260 EID1602 P3F anti-FOXO1 ChIP 1
JPK0029	GSL-JK-3260 EID1602 P3F anti-FOXO1 Input 1
JPK0030	GSL-JK-3260 EID1602 CNTL anti-FOXO1 ChIP 1
JPK0031	GSL-JK-3260 EID1602 CNTL anti-FOXO1 Input 1
JPK0032	GSL-JK-3260 EID1622 P3F anti-FOXO1 ChIP 2
JPK0033	GSL-JK-3260 EID1622 P3F anti-FOXO1 Input 2
JPK0034	GSL-JK-3260 EID1622 CNTL anti-FOXO1 ChIP 2
JPK0035	GSL-JK-3260 EID1622 CNTL anti-FOXO1 Input 2
JPK0036	GSL-JK-3260 EID1776 P3F anti-H3K27Ac ChIP 1
JPK0037	GSL-JK-3260 EID1776 P3F anti-H3K27Ac Input 1
JPK0038	GSL-JK-3260 EID1776 CNTL anti-H3K27Ac ChIP 1 
JPK0039	GSL-JK-3260 EID1776 CNTL anti-H3K27Ac Input 1 
JPK0040	GSL-JK-3260 EID1777 P3F anti-H3K27Ac ChIP 2
JPK0041	GSL-JK-3260 EID1777 P3F anti-H3K27Ac Input 2 
JPK0042	GSL-JK-3260 EID1777 CNTL anti-H3K27Ac ChIP 2 
JPK0043	GSL-JK-3260 EID1777 CNTL anti-H3K27Ac Input 2 

# Workflow


# Load packages
```{r lib}
library(tidyverse)
theme_set(theme_bw())
theme_update(plot.title = element_text(hjust = 0.5))
```

## Make up directory structure
```{bash mkdirs, eval=TRUE}
for directoryName in \
    misc \
    output \
    output/figures \
    output/fastqc/chip \
    output/fastqc/atac \
    output/fastqc/rnaseq \
    output/counts \
    output/align/chip \
    output/align/rna \
    output/peaks \
    output/gene_counts
do
    if [ ! -d ${directoryName} ]
    then
        mkdir -p ${directoryName}
    fi
done
```

## Fastqc on ChIP data
```{bash fastqc_chip, eval=TRUE}
sbatch sbatchCmds/fastqc_chip.sh
```

### Plot Fastqc output for all samples
```{r plotFastqcFun, dependson='fastqc_chip', eval=TRUE}
fastqc_plots <- function(fastqc_dir, plot_name_base) {
    adapter_file_dirs <-
        list.dirs(path = fastqc_dir,
                  full.names = TRUE,
                  recursive = FALSE)

    to_plot <-
        tibble(file = c("Adapter_Content.txt",
                        "Per_base_sequence_quality.txt"),
               column = c("Nextera_Transposase_Sequence",
                          "Mean"))

    for (i in 1:nrow(to_plot)) {
        temp_data <- NULL

        for (data_dir in adapter_file_dirs) {
            temp_data <-
                read_delim(paste0(data_dir,
                                  "/split_data/",
                                  to_plot$file[i]),
                           delim = "\t",
                           col_names = TRUE,
                           show_col_types = FALSE) %>%
                rename_all(~ str_replace_all(., " ", "_")) %>%
                rename_all(~ str_remove(., "[#']")) %>%
                select(Base, to_plot$column[i]) %>%
                mutate(Sample = str_remove(data_dir, "_001_fastqc") %>%
                          str_remove(".+\\/")) %>%
                mutate(Read = str_remove(Sample, ".+_")) %>%
                mutate(Tech = str_remove(Sample, "_.+")) %>%
                bind_rows(temp_data)
        }

        max_y <- ceiling(max(temp_data[[to_plot$column[i]]]) * 1.1)

        plot_fastqc <-
            ggplot(temp_data,
                   aes(x = Base,
                       y = get(to_plot$column[i]),
                       group = Sample,
                       color = Sample)) +
            geom_line() +
            geom_point() +
            ylab(to_plot$column[i]) +
            theme(legend.position = "none") +
            ggtitle(str_remove(to_plot$file[i], ".txt"))

        png(filename = paste0("output/figures/fastqc_",
                              plot_name_base, "_",
                              str_remove(to_plot$file[i], ".txt"),
                              ".png"),
            width = 7000,
            height = 3000,
            res = 300)
        print(plot_fastqc)
        dev.off()
    }

    input_reads <-
        read_tsv(list.files(path = "output/fastqc",
                            recursive = TRUE,
                            pattern = "Basic_Statistics.txt",
                            full.names = TRUE),
                 id = "File",
                 show_col_types = FALSE) %>%
        filter(`#Measure` == "Total Sequences") %>%
        select(-`#Measure`) %>%
        mutate(File = str_remove(File, "output/fastqc/") %>%
                    str_remove("_001_fastqc/split_data/Basic_Statistics.txt"),
               Sample = str_remove(File, "_S[0-9].+"),
               Lane = str_remove(File, ".+_L00") %>%
                    str_remove("_R[12]"),
               Read = str_replace(File, ".+_R", "R"),
               Value = as.numeric(Value)) %>%
        rename(Reads = Value)

    ggplot(input_reads,
           aes(x = Sample,
               y = Reads / 1000000,
               fill = Lane)) +
        geom_col() +
        facet_wrap(~Read, ncol = 1) +
        ylab("Reads (in millions)") +
        xlab("") +
        ggtitle("Sequenced Reads Per Sample") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1))

    ggsave(paste0("output/figures/",
                  plot_name_base,
                  "_InputReads.png"),
           height = 3000,
           width = 4000,
           units = "px")
}
```

```{r eval=FALSE}
fastqc_plots("output/fastqc/chip", "chip")
```


## Align ChIPseq libraries to zebrafish genome, index, and remove duplicates
```{bash}
mkdir /gpfs0/scratch/junkdir1337/
sbatch sbatchCmds/align_chip.sh
rm -r /gpfs0/scratch/junkdir1337
```

## call peaks
```{bash}
sbatch sbatchCmds/macs2peaks.sh
```
